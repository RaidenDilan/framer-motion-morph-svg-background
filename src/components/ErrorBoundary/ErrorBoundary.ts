import React, { ReactNode } from 'react';

export type ErrorBoundaryProps = {
  children: ReactNode;
};

export type ErrorBoundaryState = {
  error?: string;
  hasError: boolean;
};

export type ErrorBoundaryCopy = {
  errorBoundary: {
    message: string;
    reload: string;
  };
};
