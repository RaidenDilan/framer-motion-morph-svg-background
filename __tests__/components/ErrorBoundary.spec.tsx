import React, { ReactElement } from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';

import ErrorBoundary from '../../src/components/ErrorBoundary';
import {
  ErrorBoundaryProps,
  ErrorBoundaryState,
} from '../../src/components/ErrorBoundary/ErrorBoundary';

const Child = () => {
  throw new Error('Test');
};

const props: ErrorBoundaryProps & ErrorBoundaryState = {
  children: <div>Child</div>,
  error: undefined,
  hasError: false,
};

describe(__filename, () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('should render without crashing', () => {
    const { asFragment } = render(<ErrorBoundary {...props} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it(`should render error boundary component when there is an error`, async () => {
    const log = jest.spyOn(console, 'log').mockImplementation();
    const error = jest.spyOn(console, 'error').mockImplementation(() => {
      throw new Error('Test');
    });

    const { getByText } = renderProviders(
      <ErrorBoundary>
        <Child />
      </ErrorBoundary>,
    );

    expect(screen.getByTestId('errorboundary-message')).toBeVisible();

    const errorMessage = getByText('An error occurred');
    expect(errorMessage).toBeDefined();

    expect(Child).toThrowError();

    expect(log).toHaveBeenCalledTimes(1);
    expect(error).toHaveBeenCalledTimes(3);
  });
});

const renderProviders = (ui: ReactElement) => render(ui, {});
