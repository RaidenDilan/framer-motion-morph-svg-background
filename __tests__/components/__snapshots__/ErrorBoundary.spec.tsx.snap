// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/Users/Raiden/development/.boilerplates/vite-boilerplate/__tests__/components/ErrorBoundary.spec.tsx should render without crashing 1`] = `
<DocumentFragment>
  <div>
    Child
  </div>
</DocumentFragment>
`;
